can/parser.cc

```diff
@@ -63,11 +63,17 @@ bool MessageState::parse(uint64_t sec, const std::vector<uint8_t> &dat) {
       return false;
     }
 
-    // TODO: these may get updated if the invalid or checksum gets checked later
+    // Note that if counter or checksum is checked after, vals will be partially/all wrong.
+    // But we only update last_seen_nanos on success, so returned vals from query_latest are never wrong
+    LOGE("adding value for name: %s: %f", sig.name.c_str(), tmp * sig.factor + sig.offset);
     vals[i] = tmp * sig.factor + sig.offset;
+  }
+
+  // Add vals once we check all signals in the message
+  for (int i = 0; i < parse_sigs.size(); i++) {
     all_vals[i].push_back(vals[i]);
   }
```

Comment:
Fixes mismatch with Honda brake pressed since this added invalid brake switch signals (due to checksum or counter)